website/docs/r/serviceendpoint_permissions.html.markdown

```diff
@@ -0,0 +1,100 @@
+---
+layout: "azuredevops"
+page_title: "AzureDevops: azuredevops_serviceendpoint_permissions"
+description: |-
+  Manages permissions for a AzureDevOps Service Endpoint
+---
+
+# azuredevops_serviceendpoint_permissions
+
+Manages permissions for a Service Endpoint
+
+~> **Note** Permissions can be assigned to group principals and not to single user principals.
+
+## Permission levels
+
+Permission for Service Endpoints within Azure DevOps can be applied on two different levels.
+Those levels are reflected by specifying (or omitting) values for the arguments `project_id` and `serviceendpoint_id`.
+
+## Example Usage
+
+```hcl
+
+resource "azuredevops_project" "project" {
+  name               = "Sample Project"
+  work_item_template = "Agile"
+  version_control    = "Git"
+  visibility         = "private"
+  description        = "Managed by Terraform"
+}
+
+data "azuredevops_group" "project-readers" {
+  project_id = azuredevops_project.project.id
+  name       = "Readers"
+}
+
+resource "azuredevops_serviceendpoint_permissions" "root-permissions" {
+  project_id  = azuredevops_project.project.id
+  principal   = azuredevops_group.project-readers.id
+  permissions = {
+    USE               = "allow"
+    ADMINISTER        = "allow"
+    CREATE            = "allow"
+    VIEWAUTHORIZATION = "allow"
+    VIEWENDPOINT      = "allow"
+  }
+}
+
+resource "azuredevops_serviceendpoint_dockerregistry" "dockerhubregistry" {
+  project_id             = azuredevops_project.project.id
+  service_endpoint_name  = "Sample Docker Hub"
+
+  docker_username        = "sample"
+  docker_email           = "email@example.com"
+  docker_password        = "12345"
+  registry_type          = "DockerHub"
+}
+
+resource "azuredevops_serviceendpoint_permissions" "serviceendpoint-permissions" {
+  project_id         = azuredevops_project.project.id
+  principal          = azuredevops_group.project-readers.id
+  serviceendpoint_id = azuredevops_serviceendpoint_dockerregistry.dockerhubregistry.id
+  permissions = {
+    USE               = "allow"
```

Comment:
All the permission here is in upper case while the key name in [actionMap](https://github.com/microsoft/terraform-provider-azuredevops/blob/main/azuredevops/internal/service/permissions/utils/namespaces.go#L379) is in camel case, this seems a bug here.
Test script:
```
data "azuredevops_project" "project" {
  name               = "Sample Project"
}

data "azuredevops_group" "project-readers" {
  project_id = data.azuredevops_project.project.id
  name       = "Readers"
}

resource "azuredevops_serviceendpoint_permissions" "root-permissions" {
  project_id  = data.azuredevops_project.project.id
  principal   = data.azuredevops_group.project-readers.id
  permissions = {
    Use               = "allow"
    Administer        = "allow"
    Create            = "allow"
    ViewAuthorization = "allow"
    ViewEndpoint      = "allow"
  }
}

resource "azuredevops_serviceendpoint_dockerregistry" "dockerhubregistry" {
  project_id             = data.azuredevops_project.project.id
  service_endpoint_name  = "Sample Docker Hub"

  docker_username        = "sample"
  docker_email           = "email@example.com"
  docker_password        = "12345"
  registry_type          = "DockerHub"
}

resource "azuredevops_serviceendpoint_permissions" "serviceendpoint-permissions" {
  project_id         = data.azuredevops_project.project.id
  principal          = data.azuredevops_group.project-readers.id
  serviceendpoint_id = azuredevops_serviceendpoint_dockerregistry.dockerhubregistry.id
  permissions = {
    Use               = "allow"
    Administer        = "deny"
    Create            = "deny"
    ViewAuthorization = "allow"
    ViewEndpoint      = "allow"
  }
}
```


Error:
```
Error: Invalid permission [VIEWENDPOINT]

  on main.tf line 25, in resource "azuredevops_serviceendpoint_permissions" "root-permissions":
  25: resource "azuredevops_serviceendpoint_permissions" "root-permissions" {



Error: Invalid permission [Viewauthorization]

  on main.tf line 47, in resource "azuredevops_serviceendpoint_permissions" "serviceendpoint-permissions":
  47: resource "azuredevops_serviceendpoint_permissions" "serviceendpoint-permissions" {


```

Reply:
@xuzhang3 will have a look over the day

Reply:
@xuzhang3 checked the security namespace. You're correct, the capitalization for the permissions inside the documentation is incorrect

![image](https://user-images.githubusercontent.com/14177833/143195008-705892e8-fa46-47f1-8f5b-bda735c09481.png)

Reply:
@xuzhang3 fixed